msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt

```diff
@@ -2293,3 +2395,33 @@ class NativeAuthPublicClientApplicationKotlinTest : PublicClientApplicationAbstr
         assertTrue((result as SignUpError).isInvalidUsername())
     }
 }
+
+class LoggerTestCallback(private val allowPIITrueToCheck: List<String>, private val allowPIIFalseToCheck: List<String>, private val allowPII: Boolean, private val latch: CountDownLatch?) : ILoggerCallback {
+    var failCalled = false
+    override fun log(
+        tag: String?,
+        logLevel: Logger.LogLevel?,
+        message: String?,
+        containsPII: Boolean
+    ) {
+        if (containsPII) {
```

Comment:
Not too familiar with how the latch works - but if `containsPII` is false here, will the test hang since `latch?.countDown()` is never called?

Reply:
@Yuki-YuXin  this code is not complete yet. As discussed offline and in a different comment I left, we need multiple test cases that validate different values of `containsPII`. E.g. some fields should never be called, regardless of the value of containsPII.

I also think @rmccahill has a point with the above. the countdown should be called outside of the if, i.e. always when the callback is called. 

@rmccahill  the test won't hang, but it will timeout. CountDownLatch has a timeout configured, meaning that if countdown isn't called within that timeframe, the test times out.

Reply:
I first wrongly place the latch?.countDown() which has been fixed now.

Reply:
And the logic here has been updated according to the offline discussion.